ui/packages/components/src/RunDetailsV2/RunDetails.tsx

```diff
@@ -42,54 +41,29 @@ type Run = {
 export function RunDetails(props: Props) {
   const { getResult, getRun, getTrigger, pathCreator, pollInterval, rerun, runID, standalone } =
     props;
-  const [error, setError] = useState<Error>();
 
-  const [run, setRun] = useState<Run>();
-  const endedAt = run?.trace?.endedAt;
-  useEffect(() => {
-    if (endedAt) {
-      // Don't poll if the run has ended
-      return;
-    }
-
-    if (!pollInterval) {
-      if (run) {
-        // Nothing left to fetch
-        return;
+  const runRes = useQuery({
+    queryKey: ['run', runID],
+    queryFn: useCallback(() => {
+      return getRun(runID);
+    }, [getRun, runID]),
+    retry: 3,
+    refetchInterval: pollInterval,
+  });
+
+  const outputID = runRes?.data?.trace.outputID;
+  const resultRes = useQuery({
+    enabled: Boolean(outputID),
+    queryKey: ['run-result', runID],
+    queryFn: useCallback(() => {
+      if (!outputID) {
+        // Unreachable
+        throw new Error('missing outputID');
       }
 
-      withRetry(() => getRun(runID))
-        .then((data) => {
-          setRun(data);
-        })
-        .catch(setError);
-      return;
-    }
-
-    const interval = setInterval(() => {
-      withRetry(() => getRun(runID))
-        .then((data) => {
-          setRun(data);
-        })
-        .catch(setError);
-    }, pollInterval);
-
-    return () => clearInterval(interval);
-  }, [endedAt, getRun, pollInterval, runID]);
-
-  const [result, setResult] = useState<Result>();
-  const outputID = run?.trace?.outputID;
-  useEffect(() => {
-    if (!result && outputID) {
-      withRetry(() => getResult(outputID))
-        .then((data) => {
-          setResult(data);
-        })
-        .catch(() => {
-          toast.error('Failed to fetch run result');
```

Comment:
**question**: I cannot find the equivalent of this in the new React query code. What is handling this result query's error?

Reply:
We're handling React Query errors by throwing, which will be caught by the error boundary:
https://github.com/inngest/inngest/blob/fix-new-runs-loading/ui/packages/components/src/RunDetailsV2/RunDetails.tsx#L79

We swallowing errors when fetching the output. There aren't any designs for that scenario